File: app/src/main/java/im/tox/antox/callbacks/AntoxOnActionCallback.java
package im.tox.antox.callbacks;

import android.content.Context;

import im.tox.antox.utils.AntoxFriend;
import im.tox.jtoxcore.callbacks.OnActionCallback;

public class AntoxOnActionCallback implements OnActionCallback<AntoxFriend> {

    private final static String TAG = "im.tox.antox.TAG";
    private Context ctx;

    public AntoxOnActionCallback(Context ctx) {
        this.ctx = ctx;
    }

    @Override
    public void execute(AntoxFriend friend, String action) {
    }

}

File: app/src/main/scala/im/tox/antox/callbacks/AntoxOnActionCallback.scala
package im.tox.antox.callbacks

import android.content.Context
import im.tox.antox.tox.{MessageHelper, ToxSingleton}
import im.tox.antox.utils.Constants
import im.tox.tox4j.core.callbacks.FriendActionCallback

object AntoxOnActionCallback {

  private val TAG = "im.tox.antox.TAG"
}

class AntoxOnActionCallback(private var ctx: Context) extends FriendActionCallback {

  override def friendAction(friendNumber: Int, timeDelta: Int, message: Array[Byte]): Unit = {
    MessageHelper.handleMessage(ctx, friendNumber,
      ToxSingleton.getAntoxFriend(friendNumber).get.getKey(),
      new String(message, "UTF-8"),
      Constants.MESSAGE_TYPE_ACTION)
  }
}
